lib/matplotlib/tests/test_figure.py

```diff
@@ -533,3 +533,16 @@ def test_removed_axis():
     fig, axs = plt.subplots(2, sharex=True)
     axs[0].remove()
     fig.canvas.draw()
+
+
+def test_add_subplot_twotuple():
+    fig = plt.figure()
+    ax2 = fig.add_subplot(3, 2, (3, 5))
+    assert ax2.get_subplotspec().rowspan == range(1, 3)
+    assert ax2.get_subplotspec().colspan == range(0, 1)
+    ax3 = fig.add_subplot(3, 2, (4, 6))
+    assert ax3.get_subplotspec().rowspan == range(1, 3)
+    assert ax3.get_subplotspec().colspan == range(1, 2)
+    ax4 = fig.add_subplot(3, 2, (3, 6))
+    assert ax4.get_subplotspec().rowspan == range(1, 3)
+    assert ax4.get_subplotspec().colspan == range(0, 2)
```

Comment:
The (6, 3) case revealed a bug in colspan, fixed.
We don't fail for out-of-range indices because that's mapped to `gs[i:j]` which just clips i and j to within the range, consistently with list indexing.